packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -5116,14 +5088,48 @@ abstract class RenderObjectElement extends Element {
     return ancestor as RenderObjectElement;
   }
 
-  ParentDataElement<RenderObjectWidget> _findAncestorParentDataElement() {
+  ParentDataElement<ParentData> _findAncestorParentDataElement() {
     Element ancestor = _parent;
+    ParentDataElement<ParentData> result;
     while (ancestor != null && ancestor is! RenderObjectElement) {
-      if (ancestor is ParentDataElement<RenderObjectWidget>)
-        return ancestor;
+      if (ancestor is ParentDataElement<ParentData>) {
+        result = ancestor as ParentDataElement<ParentData>;
```

Comment:
I believe that either the `as` isn't necessary here anymore, or that if it is you should just be doing:

```dart
      result = ancestor as ParentDataElement<ParentData>;
      if (result != null) {
       break;

Reply:
Without the "as" I am getting:

error: A value of type 'Element' can't be assigned to a variable of type 'ParentDataElement<ParentData>'. (invalid_assignment at [flutter] lib/src/widgets/framework.dart:5096)

And in your code, the "as" would throw if ancestor is not a ParentDataElement<ParentData>, which I don't want.

Reply:
Ahh I was thinking of C#.  Nevermind

Reply:
i'm surprised you need the as, i expected this case to be handled. cc @leafpetersen